app/Main.hs

```diff
@@ -0,0 +1,116 @@
+module Main (main) where
+
+import Colourista (errorMessage, infoMessage, warningMessage)
+import Data.Foldable (forM_)
+import Data.Text (Text)
+import GHC.Exts (sortWith)
+import System.Directory (doesFileExist, getCurrentDirectory, listDirectory)
+import System.Exit (exitFailure)
+import System.FilePath (takeExtension, (</>))
+
+import Cli (ExtensionsArgs (..), runExtensionsCli)
+import Extensions (ExtensionsResult, getModuleExtentions, getPackageExtentions)
+import Extensions.Cabal (parseCabalFileExtensions)
+import Extensions.OnOff (OnOffExtension (..), mergeExtensions)
+import Extensions.Parser (parseFile)
+
+import qualified Data.Map as Map
+import qualified Data.Text as Text
+import qualified Data.Text.IO as TextIO
+
+
+main :: IO ()
+main = do
+    ExtensionsArgs{..} <- runExtensionsCli
+
+    let printExtensions :: FilePath -> IO ()
+        printExtensions = printWithCabalExtensions
+            extensionsArgsNoModules
+            extensionsArgsModuleFilePath
+
+    case extensionsArgsCabalFilePath of
```

Comment:
I have noticed that we need a strategy on contradicting arguments provided by user.
For example, what should we do if user 
```
$ extensions --no-modules --module-file-path "src/Extensions.hs"
$ extensions --no-cabal --cabal-file-path "extensions.cabal"
```
?

At the moment it ignores one of the options and provides the result just as there is only one option provided.
Do you think it would make sense to not provide anything on such output and ask to be more careful about the query?

Reply:
@vrom911 I see. It's indeed better to handle all cases explicitly to define behavior for each case clearly. Will do that 👍